engine/execution/computation/computer/uploader/uploader.go

```diff
@@ -0,0 +1,128 @@
+package uploader
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"cloud.google.com/go/storage"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/engine/execution"
+	"github.com/onflow/flow-go/module"
+	"github.com/rs/zerolog"
+)
+
+type Uploader interface {
+	Upload(computationResult *execution.ComputationResult) error
+}
+
+func NewAsyncUploader(uploader Uploader, log zerolog.Logger, metrics module.ExecutionMetrics) *AsyncUploader {
+	return &AsyncUploader{
+		unit:     engine.NewUnit(),
+		uploader: uploader,
+		log:      log.With().Str("component", "block_data_uploader").Logger(),
+		metrics:  metrics,
+	}
+}
+
+type AsyncUploader struct {
+	unit     *engine.Unit
+	uploader Uploader
+	log      zerolog.Logger
+	metrics  module.ExecutionMetrics
+}
+
+func (a *AsyncUploader) Ready() <-chan struct{} {
+	return a.unit.Ready()
+}
+
+func (a *AsyncUploader) Done() <-chan struct{} {
+	return a.unit.Done()
+}
+
+func (a *AsyncUploader) Upload(computationResult *execution.ComputationResult) error {
+	a.unit.Launch(func() {
+		a.metrics.ExecutionBlockDataUploadStarted()
+		defer a.metrics.ExecutionBlockDataUploadFinished()
+
+		err := a.uploader.Upload(computationResult)
+		if err != nil {
+			a.log.Error().Err(err).Msg("error while uploading block data")
```

Comment:
Maybe a dumb question, but considering that if the `Upload` call returns an `error`, it's probably the percolation of a `cbor` error, would we want to make sure we can log the embedded error in detail?

Reply:
we log error which should log entire structure, won't it?